packages/theme-dark-extension/style/variables.css

```diff
@@ -209,7 +209,8 @@ all of MD as it is not optimized for dense, information rich UIs.
    * theme these would go from dark to light.
    */
 
-  --jp-inverse-layout-color0: white;
+  /* Separated rbg values to allow passing into rgba. */
+  --jp-inverse-layout-color0: 155, 155, 155; /* white */
```

Comment:
We should not touch those generic variables as they may impact extensions styling. We should aim at something specific for the vim mode.

Looking at the DOM, when the VIM mode is activated the `cm-fat-cursor` is added to `CodeMirror cm-s-jupyter CodeMirror-wrap` in the top container.

I tried to add that CSS rule and it works as expected - feel free to tune the styling change this is more to point out a good CSS selector to fix the issue:

```css
.cm-fat-cursor .cm-overlay.cm-searching {
	opacity: 0.5;
}
```